gadopt/stokes_integrators.py

```diff
@@ -100,35 +106,44 @@ class StokesSolver:
     def __init__(self, z, T, approximation, bcs=None, mu=1,
                  quad_degree=6, cartesian=True, solver_parameters=None,
                  closed=True, rotational=False, J=None, constant_jacobian=False,
-                 **kwargs):
+                 iterative_2d=False, free_surface_dt=None, free_surface_variable_rho=True,
+                 free_surface_theta=0.5, **kwargs):
+
         self.Z = z.function_space()
         self.mesh = self.Z.mesh()
         self.test = fd.TestFunctions(self.Z)
         self.equations = StokesEquations(self.Z, self.Z, quad_degree=quad_degree,
                                          compressible=approximation.compressible)
         self.solution = z
         self.approximation = approximation
+
         self.mu = ensure_constant(mu)
         self.solver_parameters = solver_parameters
         self.J = J
         self.constant_jacobian = constant_jacobian
         self.linear = not depends_on(self.mu, self.solution)
 
         self.solver_kwargs = kwargs
-        u, p = fd.split(self.solution)
+        self.stokes_vars = fd.split(self.solution)  # For default stokes this is a tuple of (velocity, pressure)
```

Comment:
This way, `eta` becomes a list, right? That would require some additional changes here, but it may actually be quite handy. The logic could only fail if we added another variable I suppose, but I think it is a very nice suggestion indeed!

Reply:
Oh I forgot you could do that, I agree that's a clean way to do it!

Reply:
I've added this in now!